lib/samples/snap_geometry_edits/snap_geometry_edits.dart

```diff
@@ -574,9 +586,38 @@ class _SnapGeometryEditsState extends State<SnapGeometryEdits>
     return Column(
       crossAxisAlignment: CrossAxisAlignment.start,
       children: [
-        Text(
-          label,
-          style: Theme.of(context).textTheme.titleMedium,
+        Row(
+          mainAxisAlignment: MainAxisAlignment.spaceBetween,
+          children: [
+            Text(
+              label,
+              style: Theme.of(context).textTheme.titleMedium,
+            ),
+            Row(
+              children: [
+                const Text('Enable all'),
+                // A checkbox to enable all source settings in the category.
+                Checkbox(
+                  value: allSourceSettings.firstWhereOrNull(
+                        (snapSourceSettings) => !snapSourceSettings.isEnabled,
+                      ) ==
+                      null,
+                  onChanged: (allEnabled) {
+                    if (allEnabled != null) {
+                      allSourceSettings
+                          .map(
+                            (snapSourceSettings) =>
+                                snapSourceSettings.isEnabled = allEnabled,
+                          )
+                          .toList();
+                      // We must redraw to update the checkbox UI.
+                      setState(() {});
```

Comment:
I think, at least for consistency, the `setState` should be wrapped around the `snapSourceSettings.isEnabled = allEnabled` a couple lines up.